src/app.py

```diff
@@ -0,0 +1,48 @@
+"""
+An example stored procedure. __main__ provides an entrypoint for local development
+and testing.
+"""
+
+from snowflake.snowpark.session import Session
+from snowflake.snowpark.dataframe import col, DataFrame
+from snowflake.snowpark.functions import udf
+from src.functions import combine
```

Comment:
wouldn't it be better if we setup the settings in .vscode/settings.json

I typically use:

{
    
    "python.envFile": "${workspaceFolder}/.env",
    "files.exclude": {
        "**/.git": true,
        "Lib": true,
        "Include": true,
        "Scripts": true,
        "**/__pycache__":true
    }   
}

Reply:
I don't follow how those settings relate to the imports you highlighted

Reply:
Sorry let me see if I can explain this better. If you use settings.json file like this:

```
{
    "python.envFile": "${workspaceFolder}/.env",
    "terminal.integrated.env.linux": {"PYTHONPATH":"${workspaceFolder}/src:${workspaceFolder}/test:${PYTHONPATH}"},
    "terminal.integrated.env.osx": {"PYTHONPATH":"${workspaceFolder}/src:${workspaceFolder}/test:${PYTHONPATH}"},
    "terminal.integrated.env.windows": {
        "PYTHONPATH": "${workspaceFolder}\\src:.\\test:${PYTHONPATH}"
    },
    "python.testing.pytestArgs": [
        "test"
    ],
    "python.testing.unittestEnabled": false,
    "python.testing.pytestEnabled": true,
    "python.analysis.extraPaths": [
        "${workspaceFolder}/src"
    ]
}
```
that configures the tooling to understand that all the code is under ./src/ so you dont have to include src as part of the import which can be confusing. This also works with the debugger and whenever you open a terminal.

Reply:
If you combine that with some changes in the setup.py like:

```
from setuptools import setup, find_packages

PACKAGE_NAME = "project1"
setup(
    name=PACKAGE_NAME,
    version="0.1.0",
     # Specify the package directory
    package_dir={'': 'src'},
    # Include all files from the src directory
    package_data={PACKAGE_NAME: ['*']}
)
```

then you can just do `python setup.py bdist_wheel` and you can just renamed that whl to .zip and it will work for deployment